libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -602,7 +597,7 @@ - (void)passcodeValidatedToAuthValidation
 
 - (void)authValidationAtLaunch
 {
-    if (![SFUserAccountManager sharedInstance].isCurrentUserAnonymous && ![SFUserAccountManager sharedInstance].currentUser.credentials.accessToken && self.authenticateAtLaunch) {
+    if (self.authenticateAtLaunch &&  ![SFUserAccountManager sharedInstance].currentUser.credentials.accessToken) {
```

Comment:
In the context of `SalesforceSDKManager`, it's a matter of convenience.  The main drive of SalesforceSDKManager is to bootstrap a valid Mobile SDK context at `launch` time.  In the "no current user configured" case—which this code (somewhat opaquely) represents—the SDK manager looks to see if it's supposed to authenticate at launch.  If it is, it takes its best guess as to how to set up a user context, if one doesn't exist.  Or conversely, to bypass further context setup if one _does_ exist.

This is intentional, and would likely remain throughout any other deprecation of "current" user context management by the SDK, because it's being tasked with setting up that context.  Consumers who want full control over that context setup, would set `self.authenticateAtLaunch` to `NO`, at which point the SDK manager would bypass the setup.